QuestionAnswering/QA_ICL/data_generation/README.md

```diff
@@ -1,90 +1,56 @@
+  - [Usage](#usage)
+    - [Lexicon Generation](#lexicon-generation)
+    - [KG Schema Extraction](#kg-schema-extraction)
+    - [CSV-to-JSON Conversion of Data Request Generation Examples](#csv-to-json-conversion-of-data-request-generation-examples)
+    - [Augmentation of Data Request Generation](#augmentation-of-data-request-generation)
+    - [CSV-to-JSON Conversion of Physical Quantity Detection Examples](#csv-to-json-conversion-of-physical-quantity-detection-examples)
+  
+## Overview
 
-### Lexicon
-
-Within the context of this project, a lexicon of a class, a predicate, or an entity is defined as a collection of its text representations, which include its canonical label and surface forms. 
-
-In the main application, lexicons are used to perform:
-
-- entity linking (including fuzzy search over surface forms and vector similarity search)
-- lookup of canonical labels (useful when `rdfs:label` information is not easy to locate due to the distribution of data across different servers and namespaces)
-
-See [JSON schema for lexicons](#data-schema-of-lexicons).
-
-### KG schema extracts
-
-This directory adopts the terminology used for heterogenous graphs `G=(V, E, τ, φ)`:
-
-- For each node `v ∈ V`, its node type is `τ(v)`.
-- For each edge `(u, v) ∈ E`,
-  - its edge type is `φ(u, v)`, which is equivalent to 'predicate' or 'property' in RDF terms;
-  - its relation type is tuple `(τ(u), φ(u, v), τ(v))`.
-
-The simplified data schema captures only node types, edge types, and relation types, and disregard other schema information such as RDFS class hierarchy. 
-
-See [JSON schema for KG schema extracts](#data-schema-of-kg-schema-extracts).
-
-### Examples of data request generation from natural language questions
-
-A question can be answered so long as appropriate data is supplied. The retrieval of pertinent data is carried out by executing structured data requests that correspond to the input questions. By prompting LLMs with pairs of natural language queries and their corresponding data requests, LLMs can perform in-context learning to automatically generate data requests for unseen input questions.
-
-Data requests come in two variants: SPARQL query and function call. 
-
-See [JSON schema for examples of data request generation](#data-schema-of-data-request-generation-examples).
-
-## Installation
-
-### Prerequisites
-- python>=3.8
-- (optional but recommended) [conda](https://conda.io/projects/conda/en/latest/user-guide/install/index.html)
-
-### Steps
-
-0. If a virtual environment manager is used, create a new python environment and activate it. In the case of conda:
-   ```{bash}
-   conda create --name data-env python==3.8
-   source activate data-env
-   ```
-1. Install dependendencies.
-   ```{bash}
-   pip install -r requirements.txt
-   ```
+This directory contains Python and shell scripts to prepare three kinds of data required by Marie's backend: lexicons, graph schemas, and examples of data request generation.
```

Comment:
Reword the intro to 

The QA_ICL system requires three types of data schemas:

- Lexicons: help with entity linking
- Graph schemas: help with predicting SPARQL queries that look for information needed by users
- Examples of data request generation: provide demonstrations for how user queries ought to be mapped to structured data requests.
  - Specific examples for physical quantity recognition: provide demonstrations for how to extract mentions of physical quantities from input texts.

Users are able to generate these schemas manually by following the specifications described in the second section. There are also some script helpers to automate this process for the chemistry domain in the third section